### PR TITLE
Adding small debris

### LINKED ISSUE #70: Get data from MASTER
# Feature
## Desired behavior
We should have some debris in our dataset

## What needs to be done
- [x] Get Master
- [ ] Create data for 1 - 10cm and 10cm+
- [ ] Store in some sensible format and add to simulation

## How it can be tested
- [ ] Run the simulation and look at VTK etc.

### PR DESCRIPTION
**Pull request by gomezzz**
# Description

Added data for small debris. Following steps:
1. Took data from MASTER for spatial density of debris in ranges [1-2.5cm, 2.5-5cm, 5-7.5cm, 7.5-10cm] and computed expected number debris for altitudes between 186 and 2000km altitude (split in 500 logarithmic bins). This lead to an expected number of 598489 debris particles distributed as follows

image

2. I built Kernel Density Estimates (KDE) of inclination and eccentricity based on distribution of tracked satellites. They look as follows:

image

image

3. Sampled debris as following 
* Named debris as `DEBRIS_<size_range>_<altitude>_<idx `
* Sampled size uniform in intervals as above. Computed mass with `sat["M[kg]"] = 4 / 3 * np.pi *(sat["RADIUS[m]"])**3 * 2698.9` same as for catalog for < 10cm.
* Compute BSTAR as 
```
# Approximate BSTAR as p0 * C_D * 2*pi*r / (2 * mass * earth_radius[km])
sat["BSTAR[1 / Earth Radius]"] = (0.1570 * 2.2 * np.pi * sat["RADIUS[m]"]**2) / (2*sat["M[kg]"]*earth_radius)
```
* Sampled orbital elements with SMA in altitude bin, eccentricity and inclination based on above KDEs and w,W,E in uniform(0,2*PI) and converted to r,v

Resulting absolute velocities look as follows

image

Sizes

image

Population (red) + Debris (black)

image

BSTAR (debris)

image

BSTAR (population)
image

Remaining TODOs:
- [x] Write new population file combining base population with debris
- [x] Figure out correct BSTAR formulation. Current values are very small

## Related Pull Requests

- #136 

## Resolved Issues

- [x] fixes #70 

## How Has This Been Tested?

Not yet tested.

### CODE REVIEW

**Comment by gomezzz** [comment]
(for convenience also updated all figures)

File: notebooks/Generating_Small_Debris_Population.ipynb [hunk 1]
```diff
@@ -0,0 +1,475 @@
...
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "9a1447a2",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "num_debris_per_size = {} # will hold counts per altitude & size\n",
+    "sizes = [[1,2.5],[2.5,5.],[5,7.5],[7.5,10.0]]\n",
+    "datasets = [data_1cm,data_2cm,data_5cm,data_7cm]\n",
```
Comment (FG-TUM) [inline notebooks/Generating_Small_Debris_Population.ipynb:94]: why not directly initialize this as 
```python
datasets = [ pd.read_csv("...."), ... ]
```
Having the extra variables seems like a source for errors if we would change the bucketing at some point.

File: notebooks/Generating_Small_Debris_Population.ipynb [hunk 2]
```diff
@@ -0,0 +1,475 @@
...
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "95cf0f20",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "kde_incl = KernelDensity(kernel='gaussian',bandwidth=5e-4).fit(np.asarray(incl_base_pop).reshape(-1, 1))\n",
+    "kde_ecc = KernelDensity(kernel='gaussian',bandwidth=5e-4).fit(np.asarray(ecc_base_pop).reshape(-1, 1))"
```
Comment (FG-TUM) [inline notebooks/Generating_Small_Debris_Population.ipynb:186]: this could use a comment

**Review by FG-TUM — APPROVED** [review]

File: notebooks/Input Dataset Processing.ipynb [hunk 3]
```diff
@@ -169,7 +169,7 @@
     "            \n",
     "    # Determine Mass\n",
     "    if sat[\"RADIUS\"] > 0.01:\n",
-    "        sat[\"MASS\"] = 4 / 3 * np.pi *(sat[\"RADIUS\"])**3 * 92.937 * (sat[\"RADIUS\"]*2)**(-0.74)\n",
+    "        sat[\"MASS\"] = 4 / 3 * np.pi *(sat[\"RADIUS\"])**3 * 92.937 * (sat[\"RADIUS\"])**(-0.74)\n",
```
Comment (gomezzz) [inline notebooks/Input Dataset Processing.ipynb:172]:
```suggestion
    "        sat[\"MASS\"] = 4 / 3 * np.pi *(sat[\"RADIUS\"])**3 * 92.937 * (2*sat[\"RADIUS\"])**(-0.74)\n",
```

File: notebooks/Generating_Small_Debris_Population.ipynb [hunk 4]
```diff
@@ -0,0 +1,475 @@
...
+    "        # Sampling half size for radius and converting to m\n",
+    "        sampled_radii = np.random.uniform(low=size[0] / 2.0 / 100.0, high=size[1] / 2.0 / 100.0,size=(N_at_altitude))\n",
+    "        \n",
+    "        # Sample individual satellites\n",
+    "        for idx in range(N_at_altitude):\n",
+    "            n = \"DEBRIS_\"+str(size[0])+\"-\"+str(size[1])+\"_\"+str(altitude)+\"_\"+str(debris_index)\n",
+    "            sat = {\"COSPAR_ID\": n,\"NAME\": n, \"TYPE\": \"passive\"}\n",
+    "            sat[\"RADIUS[m]\"] = sampled_radii[idx]\n",
+    "            if sat[\"RADIUS[m]\"] > 0.01:\n",
+    "                sat[\"M[kg]\"] = 4 / 3 * np.pi *(sat[\"RADIUS[m]\"])**3 * 92.937 * (sat[\"RADIUS[m]\"])**(-0.74)\n",
```
Comment (gomezzz) [inline notebooks/Generating_Small_Debris_Population.ipynb:263]:
```suggestion
    "                sat[\"M[kg]\"] = 4 / 3 * np.pi *(sat[\"RADIUS[m]\"])**3 * 92.937 * (2*sat[\"RADIUS[m]\"])**(-0.74)\n",
```

**Pull request merged**